pkg/k8s/client.go

```diff
@@ -54,7 +55,8 @@ func createConfig(endpoint, kubeCfgPath string) (*rest.Config, error) {
 		return clientcmd.BuildConfigFromFlags("", kubeCfgPath)
 	}
 
-	config := &rest.Config{Host: endpoint}
+	userAgent := fmt.Sprintf("Cilium %s", version.Version)
```

Comment:
The user agent also needs to be added to the configuration returned on line 51 and 55